src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,156 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
 
     @Override
     public void put(K key, V value) {
-
+        putValue(hash(key), key, value);
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> current;
+        current = getNode(key);
+        return (current == null) ? null : current.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void putValue(int hash, K key, V value) {
+        Node<K, V>[] data;
+        Node<K, V> parent;
+        int i;
+        int n;
+        if ((data = table) == null || (n = table.length) == 0) {
+            data = resize();
+            n = data.length;
+        }
+        if ((parent = data[i = (n - 1) & hash]) == null) {
+            data[i] = new Node<>(hash, key, value, null);
+            size++;
+        } else {
+            Node<K, V> current;
+            K k;
+            if (parent.hash == hash && ((k = parent.key) == key
+                    || (key != null && key.equals(k)))) {
+                current = parent;
+            } else {
+                while (true) {
+                    if ((current = parent.next) == null) {
+                        parent.next = new Node<>(hash, key, value, null);
+                        size++;
+                        break;
+                    }
+                    if (current.hash == hash
+                            && ((k = current.key) == key || (key != null && key.equals(k)))) {
+                        break;
+                    }
+                    parent = current;
+                }
+            }
+            if (current != null) {
+                current.value = value;
+            }
+        }
+        if (size > threshold) {
+            resize();
+        }
+    }
+
+    private static int hash(Object key) {
+        return (key == null) ? 0 : (key.hashCode() % INITIAL_CAPACITY);
+    }
+
+    private Node<K, V>[] resize() {
+        Node<K, V>[] oldTable = table;
+        int oldCapacity = (oldTable == null) ? 0 : oldTable.length;
+        int oldThreshold = threshold;
+        int newCapacity;
+        int newThreshold;
+        if (oldCapacity > 0) {
+            newCapacity = oldCapacity * 2;
+            newThreshold = oldThreshold * 2;
```

Comment:
2 is a magic number, consider making it constant
```suggestion
            newCapacity = oldCapacity * 2;
            newThreshold = oldThreshold * 2;
```